ui/app/common/displaycontrols/observation/services/formRecordTreeBuildService.js

```diff
@@ -206,15 +219,21 @@ angular.module('bahmni.common.displaycontrol.observation')
             });
         };
 
-        self.createObsGroup = function (record) {
+        self.createObsGroup = function (record, translationData) {
             var obsGroup = {
                 "groupMembers": [],
                 "concept": {
                     "shortName": "",
                     "conceptClass": null
                 }
             };
-            obsGroup.concept.shortName = record.control.label.value;
+            var translationKey = record.control.label.translationKey;
```

Comment:
@vinishad  can you pull this (L230 -L236) into a method so that it's more readable